app/workers/reminder_worker.rb

```diff
@@ -0,0 +1,26 @@
+# TODO: Move worker logic into service
+require 'post_to_slack'
+class ReminderWorker
+  include Sidekiq::Worker
+  def perform
+
+    # First check there is an active question
+    if Question.open.any?
+      # get all active users and the active question
+      all_users = User.where(active: true)
+      todays_question = Question.open.first
+      if all_users
+        all_users.each do |user|
+          # Has this user answered?
+          if todays_question.responses.where(user: user).none?
```

Comment:
The above `if` can also be a guard clause

Reply:
It may be better to add a scope or class method to the user model to query only users who haven't answered today's question from the DB instead of loading all users from the DB and then checking if they've answered

Reply:
Trying to write this as a scope is making my head spin. 😭I can get where they haven't answered, but once I try to include the criteria of only where the question is open, it all breaks.

```
class User < ApplicationRecord
  has_many :responses

  scope :hasnt_answered, -> {
    where(active: true).left_outer_joins(:responses => :question).where(responses: {user_id: nil, question: {open: true}})
  }
end
```